aten/src/ATen/native/Attn.cpp

```diff
@@ -0,0 +1,14 @@
+#include <tuple>
+
+#include <ATen/Tensor.h>
+
+namespace at { namespace native {
+
+auto attn(Tensor const& q, Tensor const& k, Tensor const& v) -> std::tuple<Tensor, Tensor> {
+  Tensor attn = q.matmul(k.t()).tanh();
```

Comment:
It would be nice to have some error checking here, to make sure that all the inputs have the right number of dimensions at least.

Reply:
done.